proto/parca/query/v1alpha1/query.proto

```diff
@@ -256,10 +259,37 @@ message QueryRequest {
   optional SourceReference source_reference = 9;
 
   // which runtime frames to filter out, often interpreter frames like python or ruby are not super useful by default
-  optional RuntimeFilter runtime_filter = 10;
+  optional RuntimeFilter runtime_filter = 10 [deprecated = true];
 
   // invert_call_stack inverts the call stacks in the flamegraph
   optional bool invert_call_stack = 11;
+
+  // a varying set of filters to apply to the query request
+  Filters filters = 12;
+}
+
+// StackFilter is a filter for a single or multiple binary names
+message StackFilter {
+  // filter is the list of binary names to filter by
+  repeated string filter = 1;
+}
+
+// FrameFilter is a filter for a single or multiple functions
+message FrameFilter {
+  // filter is the list of function names to filter by
```

Comment:
We need the `FrameFilter` itself to be a `oneof` as well, as we already got the feature request to also be able to filter out certain functions. So we want the first `oneof` of the `FrameFilter` to be a `BinaryFrameFilter` which has a list of strings `include_binaries` (I think we should make the API an allow-list rather than a deny list, otherwise something like "show me only interpreted frames" is going to be a really odd request).